src/fibers.cc

```diff
@@ -859,6 +866,26 @@ class Fiber {
 			return uni::Return(uni::NewNumber(Isolate::GetCurrent(), Coroutine::coroutines_created()), info);
 		}
 
+		static void AdjustExternalMemoryOnFiberStart(const Fiber& fiber) {
+			int size = fiber.this_fiber->size();
+			external_bytes_used += size * GC_ADJUST;
+			int diff = external_bytes_used - external_bytes_reported;
+			if (diff > size * external_bytes_threshold_factor) {
+				uni::AdjustAmountOfExternalAllocatedMemory(fiber.isolate, diff);
+				external_bytes_reported = external_bytes_used;
+			}
+		}
+
+		static void AdjustExternalMemoryOnFiberExit(const Fiber& fiber) {
+			int size = fiber.this_fiber->size();
+			external_bytes_used -= size * GC_ADJUST;
+			int diff = external_bytes_used - external_bytes_reported;
+			if (-diff > size * external_bytes_threshold_factor) {
+				uni::AdjustAmountOfExternalAllocatedMemory(fiber.isolate, diff);
+				external_bytes_reported = external_bytes_used;
+			}
+		}
```

Comment:
These methods are similar but not identical. I didn't try to unify them because I could imagine (in the future) using a different strategy for increasing memory than for reducing memory. For example, maybe positive adjustments happen immediately, but negative adjustments are postponed a bit. On the other hand, I would hope that V8 eventually handles this kind of thing automatically.